### ISSUE TITLE
Exposing parentElement on treeAdapter.createElement 

### ISSUE THREAD

**Issue by pmdartus** [inikulin/parse5#285]
While implementing support for [custom elements in JSDOM](https://github.com/jsdom/jsdom/issues/1030#issuecomment-437083550), I am facing the following limitation. The [create an element for a token](https://html.spec.whatwg.org/multipage/parsing.html#create-an-element-for-the-token) algorithm use the *intended parent* to retrieve the associated document for a newly created node. 

In it's current implementation the JSDOM treeAdapter shares the [same document](https://github.com/jsdom/jsdom/blob/master/lib/jsdom/browser/parse5-adapter-parsing.js#L11-L13) for all the created node. However this approach doesn't work in the context of template element, since the HTMLTemplateElment content is associated with an [inert template document](https://html.spec.whatwg.org/multipage/scripting.html#associated-inert-template-document).

What do you think about extended the current createElement signature to pass the intededParent as parameter?

```ts
// The intendedParent can be undefined when this hook is invoked for the root html element.
createElement(tagName: string, namespaceURI: string, attrs: Attribute[], intendedParent?: Element): Element;
```

An alternative viable approach would be to expose in some way the [current element stack](https://github.com/inikulin/parse5/issues/237).

**Comment by inikulin** [comment]
Sorry for this taking that long. I'm alright with adding this change and will be happy to merge a PR for it if you mind to create one.

**Comment by fb55** [comment]
The alternative viable approach has been implemented in #385.
